racket/src/zuo/zuo.c

```diff
@@ -6116,163 +6116,499 @@ zuo_t *zuo_process_wait(zuo_t *pids_i) {
 }
 
 /*======================================================================*/
-/* SHA-1                                                                */
+/* SHA-256                                                              */
 /*======================================================================*/
 
-/* Based on
-    SHA-1 in C
-    By Steve Reid <sreid@sea-to-sky.net>
-  including changes by Saul Kravitz <Saul.Kravitz@celera.com>
-  and Ralph Giles <giles@ghostscript.com> */
+typedef struct zuo_sha2_ctx_t {
+    unsigned total[2];
+    unsigned state[8];
+    unsigned char buffer[64];
+    int is224;
+} zuo_sha2_ctx_t;
+
+#define ZUO_SHA256_DIGEST_SIZE 32
+
+/* rktio requires certain integer sizes, anyway: */
+typedef unsigned int uint32_sha2_t;
+typedef uintptr_t size_sha2_t;
```

Comment:
I wasn't sure if these two typedefs were right for Zuo. For the first, it would seem natural to write:
```c
typedef zuo_uint32_t uint32_sha2_t;
```
For the second, there is a `zuo_intptr_t`, but not a `zuo_uintptr_t`. On Unix, it is indeed just `intptr_t`, but on Windows:
```c
# ifdef _WIN64
typedef long long zuo_intptr_t;
# else
typedef long zuo_intptr_t;
# endif
```
Would it be better to add a `zuo_uintptr_t` and use `unsigned long long` etc., or just to keep `uintptr_t` for `size_sha2_t`?

(I definitely don't claim to understand the portability pitfalls of C integer sizes.)

Reply:
It's better to add `zuo_uintptr_t`. If I remembr correctly, trying to use `intptr_t` and `uintptr_t` created trouble with an MSVC installation that is very old but that I still use sometimes.